### PR TITLE
Make android editor controls configurable (Resolves #1509)

### LINKED ISSUE #1509: [SuperEditor][SuperReader] - Make Android overlay controls configurable through composition and use document layers
This ticket is the Android version of #1508. See that ticket for details, and replicate those, as appropriate, for Android in `SuperEditor` and `SuperReader`.

### PR DESCRIPTION
**Pull request by matthew-carroll**
Make android editor controls configurable (Resolves #1509)

### CODE REVIEW

File: super_editor/lib/src/default_editor/document_gestures_touch_android.dart [hunk 1]
```diff
@@ -1,40 +1,398 @@
...
+  /// Auto-scroller that operates based on drag handle offsets.
+  ///
+  /// The [handleAutoScroller] is set and cleared by the editor's internal Android gesture system.
+  // TODO: Try to remove this. The Android controls configuration has a handle auto scroller
+  //       when the iOS version doesn't, because the iOS version has handles built into the document
+  //       layer whereas Android displays handles in the overlay. Therefore, the overlay handles need
+  //       access to an auto-scroller. But this auto-scroller exposes an internal editor implementation
+  //       in the public scope interface. We'd rather not have this here. Once the Android composition
+  //       configuration work is done in #1509, try to remove this from the public interface.
+  // final handleAutoScroller = ValueNotifier<DragHandleAutoScroller?>(null);
```
Comment (angelosilvestre) [inline super_editor/lib/src/default_editor/document_gestures_touch_android.dart:271]: A reminder to remove the commented code.

File: super_editor/lib/src/infrastructure/platforms/mobile_documents.dart [hunk 2]
```diff
@@ -1,9 +1,137 @@
...
+  LeaderLink downstreamFocalPoint,
+);
+
+/// Builds a full-screen floating toolbar display, with the toolbar positioned near the
+/// [focalPoint], and with the toolbar attached to the given [mobileToolbarKey].
+///
+/// The [mobileToolbarKey] is used to find the toolbar in the widget tree for various purposes,
+/// e.g., within tests to verify the presence or absence of a toolbar. If your builder chooses
+/// not to build a toolbar, e.g., returns a `SizedBox()` instead of a toolbar, then the
+/// you shouldn't use the [mobileToolbarKey].
```
Comment (angelosilvestre) [inline super_editor/lib/src/infrastructure/platforms/mobile_documents.dart:83]: "then the you" -> "then you"

File: super_editor/lib/src/infrastructure/platforms/mobile_documents.dart [hunk 3]
```diff
@@ -1,9 +1,137 @@
...
+  LeaderLink focalPoint,
+);
+
+/// Builds a full-screen magnifier display, with the magnifier following the given [focalPoint],
+/// and with the magnifier attached to the given [magnifierKey].
+///
+/// The [magnifierKey] is used to find the magnifier in the widget tree for various purposes,
+/// e.g., within tests to verify the presence or absence of a magnifier. If your builder chooses
+/// not to build a magnifier, e.g., returns a `SizedBox()` instead of a magnifier, then the
+/// you shouldn't use the [magnifierKey].
```
Comment (angelosilvestre) [inline super_editor/lib/src/infrastructure/platforms/mobile_documents.dart:114]: "then the you" -> "then you"

File: super_editor/test/super_editor/mobile/super_editor_android_overlay_controls_test.dart [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        await tester.pump();
+      }
+
+      // Ensure magnifier is visible and toolbar is hidden.
+      expect(SuperEditorInspector.isMobileMagnifierVisible(), isTrue);
+      expect(SuperEditorInspector.isMobileToolbarVisible(), isFalse);
+
+      // Resolve the gesture so that we don't have pending gesture timers.
+      await gesture.up();
+      await tester.pump(const Duration(milliseconds: 100));
```
Comment (angelosilvestre) [inline super_editor/test/super_editor/mobile/super_editor_android_overlay_controls_test.dart:40]: Is there a reason why we need 100ms?

File: super_editor/test/super_editor/supereditor_component_selection_test.dart [hunk 5]
```diff
@@ -549,34 +549,19 @@ Future<void> _pumpEditorWithUnselectableHrsAndFakeToolbar(
...
-        ),
-      ),
-    ),
-  );
+  await tester //
+      .createDocument()
+      .withCustomContent(paragraphThenHrThenParagraphDoc())
+      .withGestureMode(debugDefaultTargetPlatformOverride == TargetPlatform.android
+          ? DocumentGestureMode.android
+          : DocumentGestureMode.iOS)
```
Comment (angelosilvestre) [inline super_editor/test/super_editor/supereditor_component_selection_test.dart:557]: Isn't this the default?

**Review by angelosilvestre — APPROVED** [review]
LGTM with a few minor comments.

File: super_editor/test/super_editor/mobile/super_editor_android_overlay_controls_test.dart (hunk 4 above)
Comment (matthew-carroll) [reply super_editor/test/super_editor/mobile/super_editor_android_overlay_controls_test.dart:40]: Switched to `kTapMinTime`

File: super_editor/test/super_editor/supereditor_component_selection_test.dart (hunk 5 above)
Comment (matthew-carroll) [reply super_editor/test/super_editor/supereditor_component_selection_test.dart:557]: That's what was already there, but all tests pass without those lines so I'll delete them.

**Pull request merged**